ArduSub/GCS_Mavlink.cpp

```diff
@@ -115,6 +115,95 @@ void GCS_MAVLINK_Sub::send_scaled_pressure3()
         sub.celsius.temperature() * 100);
 }
 
+
+void GCS_MAVLINK_Sub::send_attitude_quaternion()
+{
+    const AP_AHRS &ahrs = AP::ahrs();
+    const Vector3f omega = ahrs.get_gyro();
+
+    Quaternion q;
+    q.from_rotation_matrix(ahrs.get_rotation_body_to_ned());
+
+    mavlink_msg_attitude_quaternion_send(
+        chan,
+        AP_HAL::millis(),
+        q.q1,
+        q.q2,
+        q.q3,
+        q.q4,
+        omega.x,
+        omega.y,
+        omega.z);
+}
+
+
+void GCS_MAVLINK_Sub::send_pwms()
+{
+  mavlink_channel_t chan = MAVLINK_COMM_0;
+  uint32_t time_boot_ms = AP_HAL::millis();
+  uint8_t port = 0;
+  uint8_t rssi = 0;
+
+  mavlink_msg_rc_channels_scaled_send(
+      chan,
+      time_boot_ms,
+      port,
+      sub.motors.get_raw_command(AP_MOTORS_MOT_1),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_2),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_3),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_4),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_5),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_6),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_7),
+      sub.motors.get_raw_command(AP_MOTORS_MOT_8),
+      rssi
+  );
+}
+
+
+void GCS_MAVLINK_Sub::send_control_system_state()
+{
+    const AP_AHRS &ahrs = AP::ahrs();
+    const Vector3f acc_ef = ahrs.get_accel_ef();
+
+    Quaternion q_btn;
+    Quaternion q_acc;
+
+    q_btn.from_rotation_matrix(ahrs.get_rotation_body_to_ned());
+
+    q_acc.q1 = 0.0f;
+    q_acc.q2 = acc_ef.x;
+    q_acc.q3 = acc_ef.y;
+    q_acc.q4 = acc_ef.z + GRAVITY_MSS;
+
+    q_acc = q_btn.inverse() * q_acc * q_btn;
+
+    mavlink_channel_t chan = MAVLINK_COMM_0;
+    mavlink_control_system_state_t control_system_state = {};
+
+    control_system_state.time_usec = AP_HAL::micros64(); /*< [us] Timestamp (UNIX Epoch time or time since system boot). The receiving end can infer timestamp format (since 1.1.1970 or since system boot) by checking for the magnitude the number.*/
```

Comment:
control_system_state message is missused to send linear and angular accelerations.